File: core/src/main/java/com/canyon/commons/HardwareUtils.java
package com.canyon.commons;

import java.net.InetAddress;
import java.net.UnknownHostException;

public class HardwareUtils {
    public static long getPID() {
        String processName = java.lang.management.ManagementFactory.getRuntimeMXBean().getName();
        if (StringUtils.isNotEmpty(processName)) {
            return Long.parseLong(processName.split("@")[0]);
        }
        return 0L;
    }

    public static String getComputerName() {
        try {
            return InetAddress.getLocalHost().getHostName();
        } catch (UnknownHostException e) {
            return "";
        }
    }
}

File: core/src/main/kotlin/com/canyon/commons/HardwareUtils.kt
package com.canyon.commons

import java.net.InetAddress

object HardwareUtils {
    /**
     * 获取当前程序的PID
     */
    fun getPID(): Long {
        val processName = java.lang.management.ManagementFactory.getRuntimeMXBean().name
        if (processName != null && processName.isNotEmpty()) {
            return try {
                java.lang.Long.parseLong(processName.split("@")[0])
            } catch (e: Exception) {
                0
            }

        }
        return 0
    }

    /**
     * 获取当前计算机名称
     */
    fun getComputerName(): String {
        return InetAddress.getLocalHost().hostName
    }
}